tests/flow/test_query_validation.py

```
@@ -502,6 +502,35 @@ def test34_self_referential_properties(self):
+                """CREATE (root:Root {name: 'x'}),
+                          (child1:TextNode {var: floor(any(v4 IN [2] WHERE child1 = [root IN keys(root)]))}),
+                          (child2:IntNode {v0: 0})""",
+                # minimal forms of the same thing
+                """CREATE (a:L {v: keys(a)})""",
+                """CREATE (a:L {v: [x IN keys(a) | x]})""",
+                """CREATE (a:L {v: any(x IN [1] WHERE a.q = 1)})"""]
+
+        for query in queries:
+            try:
+                self.graph.query(query)
+                assert(False)
```

Ruff (0.16.0)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)